### PR TITLE
Chore: transfer LiveLogs.test to testing-library

### LINKED ISSUE #46197: Explore: Convert Enzyme tests in LiveLogs.test.tsx to use RTL
- Three tests around rendering different types of logs
- One utility function about log creation

https://github.com/grafana/grafana/blob/main/public/app/features/explore/LiveLogs.test.tsx

### PR DESCRIPTION
**Pull request by L-M-K-B**
Fixes #46197

### CODE REVIEW

File: public/app/features/explore/LiveLogs.test.tsx [hunk 1]
```diff
@@ -19,68 +40,41 @@ describe('LiveLogs', () => {
...
-    expect(wrapper.contains('log message 3')).not.toBeTruthy();
-    expect(wrapper.find('LogMessageAnsi')).toHaveLength(2);
-    expect(wrapper.find('LogMessageAnsi').first().prop('value')).toBe('log message \u001B[31m2\u001B[0m');
-    expect(wrapper.find('LogMessageAnsi').last().prop('value')).toBe('log message \u001B[31m3\u001B[0m');
+    const logList = screen.getAllByTestId('ansiLogLine');
+    expect(logList).toHaveLength(2);
+    expect(logList[0]).toHaveAttribute('style', 'color: rgb(204, 0, 0);');
+    expect(logList[1]).toHaveAttribute('style', 'color: rgb(204, 0, 0);');
   });
 });
```
Comment (L-M-K-B) [inline public/app/features/explore/LiveLogs.test.tsx:80]: The following function is used in other unit tests as well e.g. SecondaryActions. That's why I would like to move it to some utils file or create a file for it. What do you think?

File: public/app/features/explore/LiveLogs.tsx [hunk 2]
```diff
@@ -136,7 +136,8 @@ class LiveLogs extends PureComponent<Props, State> {
                 // This is triggered on every update so on every new row. It keeps the view scrolled at the bottom by
                 // default.
                 if (this.liveEndDiv && !isPaused) {
-                  this.scrollContainerRef.current?.scrollTo(0, this.scrollContainerRef.current.scrollHeight);
+                  // scrollTo needs to be optional as it is not implemented in JSDOM
+                  this.scrollContainerRef.current?.scrollTo?.(0, this.scrollContainerRef.current.scrollHeight);
```
Comment (gelicia) [inline public/app/features/explore/LiveLogs.tsx:140]: Nit: Maybe check if the function exists in the if statement here instead?

File: public/app/features/explore/LiveLogs.test.tsx [hunk 3]
```diff
@@ -19,68 +40,41 @@ describe('LiveLogs', () => {
       />
     );
 
-    expect(wrapper.contains('log message 1')).toBeTruthy();
-    expect(wrapper.contains('log message 2')).toBeTruthy();
-    expect(wrapper.contains('log message 3')).toBeTruthy();
-  });
+    expect(screen.getByRole('cell', { name: 'log message 1' })).toBeInTheDocument();
+    expect(screen.getByRole('cell', { name: 'log message 2' })).toBeInTheDocument();
+    expect(screen.getByRole('cell', { name: 'log message 3' })).toBeInTheDocument();
```
Comment (gelicia) [inline public/app/features/explore/LiveLogs.test.tsx:45]: This wasnt part of the original test, but we might want to check that 4, 5 and 6 are not visible here as well.

File: public/app/features/explore/LiveLogs.tsx (hunk 2 above)
Comment (L-M-K-B) [reply public/app/features/explore/LiveLogs.tsx:140]: I like the idea and it seems to work.

**Review by gelicia — APPROVED** [review]
LGTM!

**Pull request merged**